pkg/dataflatten/jwt.go

```diff
@@ -0,0 +1,49 @@
+package dataflatten
+
+import (
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/golang-jwt/jwt/v5"
+)
+
+// JWTFile adds support for the kolide_jwt table, which allows parsing
+// a file containing a JWT. Note that the kolide_jwt table does not handle
+// verification - this is a utility table for convenience.
+func JWTFile(file string, opts ...FlattenOpts) ([]Row, error) {
+	return flattenJWT(file, opts...)
+}
+
+func flattenJWT(path string, opts ...FlattenOpts) ([]Row, error) {
+	results := map[string]interface{}{}
+	jwtFH, err := os.Open(path)
+	if err != nil {
+		return nil, fmt.Errorf("unable to access file: %w", err)
+	}
+
+	defer jwtFH.Close()
+
+	tokenRaw, err := io.ReadAll(jwtFH)
+	if err != nil {
+		return nil, fmt.Errorf("unable to read JWT: %w", err)
+	}
+
+	// attempt decode into the generic (default) MapClaims struct to ensure we capture
+	// any claims data that might be useful
+	token, _, err := new(jwt.Parser).ParseUnverified(string(tokenRaw), jwt.MapClaims{})
+	if err != nil {
+		return nil, fmt.Errorf("unable to parse JWT: %w", err)
+	}
+
+	claims, ok := token.Claims.(jwt.MapClaims)
+	if !ok {
+		return nil, fmt.Errorf("JWT has no parseable claims")
+	}
+
+	for k, v := range claims {
+		results[k] = v
+	}
```

Comment:
Do we need `results` or can we pass `claims` straight to Flatten?

And is there any more data we should pass back? Like, this is `claims`, but is there more? I kinda want to return
```
struct {
  Claims: map[string]any
}
```

Because then if we add validate, or file metadata, or ???, we won't need to change the structure.

Reply:
I can try that out! we could additionally include the header (likely containing the signing algo) but in general its just:
| Header | Payload | Signature |
|--------|--------|--------|
| algo | claims | signature |
| type (=JWT) |  | |

Reply:
Algo seems interesting. An issued timestamp if one exists...

And someday I think it would be super interesting to get a validation in, and that would give us a place to hang it. I guess we could preemptively add `validated: false` or similar.

Reply:
we can't pass claims to flatten directly but that will work for the header and signature. the registered claim for timestamp will be at `claims->iat` if included. and I will add a validated `false` for clarity here, thank you!

Reply:
PR description is updated with the new format discussed here